packages/nextcloud/lib/src/webdav/client.dart

```diff
@@ -232,29 +232,36 @@ class WebDavClient {
       );
 
   /// Gets the content of the file at [path].
+  ///
+  /// If the response is empty the file will be created with no data.
   Future<void> getFile(
     PathUri path,
     File file, {
     void Function(double progress)? onProgress,
   }) async {
     final response = await getStream(path);
     final contentLength = response.contentLength;
-    if (contentLength != null && contentLength > 0) {
-      final sink = file.openWrite();
-      final completer = Completer<void>();
-      var downloaded = 0;
-
-      response.stream.listen((chunk) async {
-        sink.add(chunk);
-        downloaded += chunk.length;
-        onProgress?.call(downloaded / contentLength);
-        if (downloaded >= contentLength) {
-          completer.complete();
-        }
-      });
-      await completer.future;
-      await sink.close();
+
+    if (contentLength == null || contentLength <= 0) {
+      await file.create();
+      onProgress?.call(1);
+      return;
     }
```

Comment:
Did you test if just opening the sink, adding no elements and then closing it also creates the file? Then we wouldn't need an special handling

Reply:
Yep this is working but we would still need to emit the `1` as the progress again creating a separate condition:
Is this really easier to read?

```
  Future<void> getFile(
    PathUri path,
    File file, {
    void Function(double progress)? onProgress,
  }) async {
    final response = await getStream(path);
    final contentLength = response.contentLength;

    final sink = file.openWrite();

    if (contentLength != null && contentLength > 0) {
      final completer = Completer<void>();
      var downloaded = 0;

      response.stream.listen((chunk) async {
        sink.add(chunk);
        downloaded += chunk.length;
        onProgress?.call(downloaded / contentLength);
        if (downloaded >= contentLength) {
          completer.complete();
        }
      });

      await completer.future;
    } else {
      onProgress?.call(1);
    }

    await sink.close();
  }
```

Reply:
I forgot about that, then this is indeed cleaner.